lib/viewmodels/map/map_pin_view_model.dart

```diff
@@ -88,7 +175,23 @@ class MapPinViewModel extends AutoDisposeAsyncNotifier<List<MapPinDetails>> {
       ),
     );
 
-    return posts.map(_toMapPinDetails).toList();
+    return posts
+        .map(
+          (post) => _toMapPinDetails(
+            post,
+            callback: context != null
+                ? () => showDialog(
+                      context: context!,
```

Comment:
Clicking on the button for `my posts` brings us on the profile. I'm therefore wondering if clicking on the button on a challenge should bring us to the challenge page. I'm curious to have you opinion on this idea, what do you think of this?

Reply:
It was also my initial idea, but to be honest I didn't find a way to navigate to this page. Triggering a navigation of the bottom navigation bar revealed itself harder than planned and started to fall out of the scope of the PR.

Reply:
Yes ok this makes a lot of sense, I understand better. This is fine as you did! :)